django/db/models/options.py

```diff
@@ -576,14 +582,13 @@ def get_all_field_names(self):
     def get_all_related_objects(self, local_only=False, include_hidden=False,
                                 include_proxy_eq=False):
 
-        include_parents = local_only is False
+        include_parents = True if local_only is False else PROXY_PARENTS
```

Comment:
I'm not sure what you by the "Where does PROXY_PARENTS change?" question.

The logic of PROXY_PARENTS isn't tested. The logic is quite extensively tested indirectly by the different get_fields() calls, but I'll add a direct test for it.